tlsutil/config_test.go

```diff
@@ -497,16 +471,241 @@ func TestConfigurator_IncomingHTTPS_TLSMinVersion(t *testing.T) {
+}
+
+func TestConfigurator_IncomingRPCConfig(t *testing.T) {
+	c := NewConfigurator(&Config{})
+	tlsConf, err := c.IncomingRPCConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.NoClientCert, tlsConf.ClientAuth)
+
+	c.Update(&Config{VerifyIncoming: true, CAFile: "../test/ca/root.cer", CertFile: "../test/key/ourdomain.cer", KeyFile: "../test/key/ourdomain.key"})
+	tlsConf, err = c.IncomingRPCConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.RequireAndVerifyClientCert, tlsConf.ClientAuth)
+
+	c.Update(&Config{VerifyIncomingRPC: true, CAFile: "../test/ca/root.cer", CertFile: "../test/key/ourdomain.cer", KeyFile: "../test/key/ourdomain.key"})
+	tlsConf, err = c.IncomingRPCConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.RequireAndVerifyClientCert, tlsConf.ClientAuth)
+
+	c.Update(&Config{VerifyIncomingHTTPS: true, CAFile: "../test/ca/root.cer", CertFile: "../test/key/ourdomain.cer", KeyFile: "../test/key/ourdomain.key"})
+	tlsConf, err = c.IncomingRPCConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.NoClientCert, tlsConf.ClientAuth)
+}
+
+func TestConfigurator_IncomingHTTPSConfig(t *testing.T) {
+	c := NewConfigurator(&Config{})
+	tlsConf, err := c.IncomingHTTPSConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.NoClientCert, tlsConf.ClientAuth)
+
+	c.Update(&Config{VerifyIncoming: true, CAFile: "../test/ca/root.cer", CertFile: "../test/key/ourdomain.cer", KeyFile: "../test/key/ourdomain.key"})
+	tlsConf, err = c.IncomingHTTPSConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.RequireAndVerifyClientCert, tlsConf.ClientAuth)
+
+	c.Update(&Config{VerifyIncomingHTTPS: true, CAFile: "../test/ca/root.cer", CertFile: "../test/key/ourdomain.cer", KeyFile: "../test/key/ourdomain.key"})
+	tlsConf, err = c.IncomingHTTPSConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.RequireAndVerifyClientCert, tlsConf.ClientAuth)
+
+	c.Update(&Config{VerifyIncomingRPC: true, CAFile: "../test/ca/root.cer", CertFile: "../test/key/ourdomain.cer", KeyFile: "../test/key/ourdomain.key"})
+	tlsConf, err = c.IncomingHTTPSConfig()
+	require.NoError(t, err)
+	require.Equal(t, tls.NoClientCert, tlsConf.ClientAuth)
+}
+
+func TestConfigurator_OutgoingRPCConfig(t *testing.T) {
+	c := NewConfigurator(&Config{})
+	tlsConf, err := c.OutgoingRPCConfig()
+	require.NoError(t, err)
+	require.Nil(t, tlsConf)
+
+	c.Update(&Config{VerifyOutgoing: true})
+	tlsConf, err = c.OutgoingRPCConfig()
+	require.Error(t, err)
+
+	c.Update(&Config{VerifyOutgoing: true, CAFile: "../test/ca/root.cer"})
+	tlsConf, err = c.OutgoingRPCConfig()
+	require.NoError(t, err)
+
+	c.Update(&Config{VerifyOutgoing: true, CAPath: "../test/ca_path"})
+	tlsConf, err = c.OutgoingRPCConfig()
+	require.NoError(t, err)
+}
+
+func TestConfigurator_OutgoingTLSConfigForChecks(t *testing.T) {
+	c := NewConfigurator(&Config{})
+	tlsConf, err := c.OutgoingTLSConfigForCheck("")
+	require.NoError(t, err)
+	require.False(t, tlsConf.InsecureSkipVerify)
```

Comment:
Why would we default to verifying certs if `EnableAgentTLSForChecks` is false?

Reply:
`EnableAgentTLSForChecks` only says that the agent tls config should be used to create `tls.Config`. As you can see here: https://github.com/hashicorp/consul/pull/5374/files#diff-6ea70e0a4de93ebcb83000a028edc163L2309 the default for an empty config is `tls.Config{InsecureSkipVerify: false}`.